dlt-connector/src/data/proto/3_3/GradidoDeferredTransfer.ts

```diff
@@ -10,8 +14,11 @@ import { TimestampSeconds } from './TimestampSeconds'
 // seed must be long enough to prevent brute force, maybe base64 encoded
 // to own account
 // https://www.npmjs.com/package/@apollo/protobufjs
-// eslint-disable-next-line no-use-before-define
-export class GradidoDeferredTransfer extends Message<GradidoDeferredTransfer> {
+export class GradidoDeferredTransfer
```

Comment:
And how is singing the storno transaction?

Reply:
I can imaging you create a type of transaction for a created link by the user.

In the case of no activation you need a kind of storno transaction, but for signing this storno without any user interaction you need keys of a technical user. 

What do you think about to use for such a "quasi" technical user the keypair of the community, because the system of the community will trigger this storno transaction?

Reply:
### procedure for DeferredTransfer
- user create DeferredTransfer
- seed will generated and recipient address is public key generated from seed
- user sign with his own private key
- give seed to generate recipient key pair to another user
- other used use seed for claiming: 
  - generate key pair from seed
  - create transfer transaction (gradidoSend) from DeferredTransfer recipient to his recipient address
  - sign with generated key pair

### If DeferredTransfer was not claimed in timeout
To claim it the seed is needed which should be only known by user how create the deferred transfer transaction and whom he gave it. 
We could store the seed in the db.. and let sending user auto-claim it after timeout or every admin how manipulate the code the direct the transaction to himself. Or simply take the seed out of the db to claim it. So this would weakening the security which I try to archive.
If DeferredTransfer send via Community are cancelled by community, how are DeferredTransfer cancelled send directly from user? You know it is also a crypto account planned, where the community don't has the keys. 

### Alternative
We don't use DeferredTransfer at all and let Community Server Handling all the logic and create a simply transferTransaction (gradidoSend) if recipient claim his gradidos send by link.


My goal with this transaction type was to create a secure alternative to sending Gradidos via link, which can really only be redeemed by the creator and the people he has given the link to. Where the key is not stored in the database and no admin can redeem it for themselves.

Reply:
It sounds quite complex and far ahead of our current requirements, right?
So please let us go on step by step and keep your ideas and solutions in mind or better in concept-documents than in to mergable and deliverable code. 
Because it is very hard to follow a solution for a perhaps-in-futuer-possible-requirement

Reply:
Link Transaktionen sind schon vorhanden, das ist die Blockchain Umsetzung davon. 
Aber es ist noch unfertig, das sind nur die ersten Teile davon.
Es ist auch im Node Server noch nicht getestest. 
Ich kann eigentlich erst ein Konzept Dokument daraus machen, nachdem ich es programmiert und getestet habe. Erst bei der Arbeit damit zeigen sich Schwächen die ich erst beheben möchte bevor ich das in ein Konzept Dokument packe.

Wenn ich das jetzt komplett rausnehme entsteht eine Lücke im OneOf Array von TransactionBody und wenn ich die Lücke schließe ist es nicht mehr kompatible mit dem Node Server Code. 
Und aus dem Node Server will ich den ganzen Code dafür ungern wieder entfernen, dann fange ich ja von vorne an, wenn ich das endlich final implementiere.

Reply:
Das Problem ist das es mir sehr schwer fällt Konzepte auf Papier zu bringen, immer wenn ich das versucht habe, sank meine Konzentration sehr schnell und ich konnte nicht mehr weiter machen. Für mich ist es viel einfacher das erstmal zu programmieren und auszuprobieren. Mit dem DCI Modell habe ich das Gefühl das ich daraus dann auch leichter Diagramme erstellen kann um das ganze nochmal zu verbildlichen.

Reply:
Hey Dario, kein Problem...
Hast du dir die Konzepte:
- https://github.com/gradido/gradido/blob/master/docu/Concepts/BusinessRequirements/UC_Send_Contribution.md
- https://github.com/gradido/gradido/blob/master/docu/Concepts/BusinessRequirements/UC_Send_Users_Gradido.md
mal angeschaut?
Da ist schon sehr viel vorausgedacht, was dieses Thema angeht.
Wir können aber auch gerne in einem Pairing uns darüber austauschen... 
daher schließe ich diese Conversation

Reply:
Ich kenne das Konzept: UC_Send_Users_Gradido.md,
mein Konzept basiert darauf.
Das andere habe ich mir im Detail nicht angeschaut, da ich es eh nicht sinnvoll auf Blockchain-Ebene abbilden kann. Das würde ich einfach komplett im Community Server belassen und der macht dann beim einlösen eine normale Schöpfungstransaktion für die Blockchain.